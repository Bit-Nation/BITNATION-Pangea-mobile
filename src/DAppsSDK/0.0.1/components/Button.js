// @flow

import React, { Component } from 'react';
import {
  TouchableOpacity,
  View,
  Text,
} from 'react-native';

import styles from '../../../global/Styles';

type Props = {
  /**
   * @desc Props that should be passed as they are to backed native component.
   */
  nativeProps: Object,
  /**
   * @desc Name of predefined type of component.
   */
  type: string,
  /**
   * @desc Style object.
   */
  style?: Object,
  /**
   * @desc Title of the button.
   */
  title: string,
  /**
   * @desc Style of button title.
   */
  titleStyle?: Object,
  /**
   * @desc Set if button should be disabled.
   */
  disabled: boolean,
  /**
   * @desc Callback to be called when button is pressed.
   */
  onPress: () => any,
}

export default class Button extends Component<Props> {
  static validNativeProps = [
    'disabled',
  ];

  static callbackProps = [
    'onPress',
  ];

  static customProps = [
    'type',
    'title',
    'style',
    'titleStyle',
  ];

  static defaultProps = {
    type: 'transparent',
    disabled: false,
    title: '',
    onPress: () => undefined,
  };

  rootStyleForType(type: string) {
    switch (type) {
      case 'transparent':
        return styles.baseButton;
      case 'action':
        return styles.actionButton;
      case 'custom':
        return null;
      default:
        return undefined;
    }
  }

  titleStyleForType(type: string) {
    switch (type) {
      case 'transparent':
        return styles.buttonTitle;
      case 'action':
        return styles.actionButtonTitle;
      case 'custom':
        return null;
      default:
        return undefined;
    }
  }

  render() {
    const typeStyle = this.rootStyleForType(this.props.type);
    if (typeStyle === undefined) {
      console.warn(`Invalid value '${this.props.type}' for 'type' property of 'Button' component`);
    }

    return (

      <View
        style={[
          typeStyle,
          this.props.style,
        ]}
      >
        <TouchableOpacity
          testID='Touchable'
          {...this.props.nativeProps}
          style={[styles.buttonContainer]}
          onPress={() => this.props.onPress()}
        >
          {
            <Text style={[this.titleStyleForType(this.props.type), this.props.nativeProps.disabled ? styles.disabledButtonTitle : null, this.props.titleStyle]}>
              {this.props.title}
            </Text>
          }
        </TouchableOpacity>
      </View>
    );
  }
}
