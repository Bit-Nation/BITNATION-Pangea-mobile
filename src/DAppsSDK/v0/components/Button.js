// @flow

import React, { Component } from 'react';

import { Button as ReactNativeButton } from 'react-native';
import styles from '../../../global/Styles';

type Props = {
  /**
   * @desc Props that should be passed as they are to backed native component.
   */
  nativeProps: Object,
  /**
   * @desc Name of predefined type of component.
   */
  type: string,
  /**
   * @desc Style object.
   */
  style?: Object
}

const types = [
  'transparent',
  'action',
];


export default class Button extends Component<Props> {
  static validNativeProps = [
    'title',
    'color',
    'disabled',
  ];

  static callbackProps = [
    'onPress',
  ];

  static customProps = [
    'type',
    'style',
  ];

  static defaultProps = {
    type: 'body',
  };

  styleForType(type: string) {
    if (types.includes(type) === false) {
      return null;
    }

    return styles[type];
  }

  render() {
    const typeStyle = this.props.type ? this.styleForType(this.props.type) : null;
    if (this.props.type != null && typeStyle == null) {
      console.warn(`Invalid value '${this.props.type}' for 'type' property of 'Button' component`);
    }

    return (
      <ReactNativeButton
        {...this.props.nativeProps}
        style={[typeStyle, this.props.style]}
      />
    );
  }
}
