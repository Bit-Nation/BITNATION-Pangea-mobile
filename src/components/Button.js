import React, { Component } from 'react';
import { TouchableOpacity, View } from 'react-native';
import { MediaQueryStyleSheet } from 'react-native-responsive';
import PropTypes from 'prop-types';

import Text from './Text';
import Colors from '../global/Colors';

export default class Button extends Component {

  render() {
    const { style, children, onPress, enabled, ...props } = this.props;

    return (
      <View style={[
        styles.baseButton,
        enabled ? styles.enabledButton : styles.disabledButton,
        style]
      } {...props}>
        {
          enabled ?
            <TouchableOpacity style={[styles.container]} onPress={onPress}>
              {children || this._renderTitle()}
            </TouchableOpacity>
            :
            <View style={styles.container}>
              {children || this._renderTitle()}
            </View>
        }

      </View>
    );
  }

  _renderTitle() {
    return (
      <Text buttonTitle={this.props.enabled} disabledButtonTitle={!this.props.enabled}>
        {this.props.title}
      </Text>
    );
  }

}

Button.propTypes = {
  enabled: PropTypes.bool
};

Button.defaultProps = {
  enabled: true,
};

const styles = MediaQueryStyleSheet.create({
  baseButton: {
    borderRadius: 15,
    height: 30,
    justifyContent: 'center',
  },
  enabledButton: {
    backgroundColor: Colors.buttonColor,
  },
  disabledButton: {
    backgroundColor: Colors.disabledButtonColor,
  },
  container: {
    marginLeft: 13,
    marginRight: 13,
  }
});