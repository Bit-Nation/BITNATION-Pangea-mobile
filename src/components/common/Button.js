// @flow

import * as React from 'react';
import {
  TouchableOpacity,
  View,
  Text,
} from 'react-native';
import { MediaQueryStyleSheet } from 'react-native-responsive';

import GlobalStyles from '../../global/Styles';

type Props = {
  /**
   * @desc Style to be applied on top of default styles.
   */
  style?: any,

  styleTitle?: any,
  /**
   * @desc Component(s) to be rendered inside a button.
   */
  children?: React.Node,
  /**
   * @desc Callback to be called on button press.
   */
  onPress?: () => any,
  /**
   * @desc Flag that determines if button is enabled.
   */
  enabled?: boolean,
  /**
   * @desc Title to show on button.
   */
  title?: string,
}

/**
 * @desc Component that renders common button.
 * @return {React.Component} A component.
 */
const Button = ({
  style, children, onPress, enabled, styleTitle, title, ...props
}: Props) => {
  const styles = MediaQueryStyleSheet.create({
    ...GlobalStyles,
  });

  return ((
    <View
      style={[
          styles.baseButton,

          style,
        ]}
      {...props}
    >
      <TouchableOpacity testID='Touchable' disabled={!enabled} style={[styles.buttonContainer]} onPress={onPress}>
        {
        children ||
        <Text style={[styles.buttonTitle, styleTitle, (enabled === false) && styles.disabledButtonTitle]}>
          {title}
        </Text>
      }
      </TouchableOpacity>
    </View>
  ));
};

Button.defaultProps = {
  style: undefined,
  children: undefined,
  enabled: true,
  onPress: () => null,
  title: '',
};

export default Button;
