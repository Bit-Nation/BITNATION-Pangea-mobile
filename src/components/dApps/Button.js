import React, { Component } from 'react';

import { Button as ReactNativeTextButton } from 'react-native';

export default class Button extends Component {
  static validNativeProps = [
    'style',
    'title',
    'color',
  ];

  static stateBasedProps = {
    disabled: 'get',
  };

  static callbackProps = [
    'onPress',
  ];

  render() {
    return (
      <ReactNativeTextButton {...this.props} />
    );
  }
}
