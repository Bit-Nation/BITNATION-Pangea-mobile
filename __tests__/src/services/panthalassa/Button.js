import React from 'react';
import { TouchableOpacity } from 'react-native';
import { shallow } from 'enzyme';
import Button from '../../../../src/components/common/Button';

describe('Testing Button component', () => {
  test('Press works', () => {
    const onPress = jest.fn();
    const wrapper = shallow(
      <Button onPress={onPress} enabled={true}/>
    );
    expect(wrapper).toMatchSnapshot();
    wrapper.simulate('press');
    expect(onPress).toHaveBeenCalledTimes(1);
  });
});