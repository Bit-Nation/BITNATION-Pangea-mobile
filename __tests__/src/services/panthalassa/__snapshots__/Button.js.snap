// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Button component Press works 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    enabled={true}
    onPress={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <TouchableOpacity
        activeOpacity={0.2}
        onPress={[MockFunction]}
        style={
          Array [
            Object {
              "marginLeft": 13,
              "marginRight": 13,
            },
          ]
        }
      >
        <_Text
          buttonTitle={true}
          disabledButtonTitle={false}
        />
      </TouchableOpacity>,
      "style": Array [
        Object {
          "borderRadius": 15,
          "height": 30,
          "justifyContent": "center",
        },
        Object {
          "backgroundColor": "#1C497E",
        },
        undefined,
      ],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeOpacity": 0.2,
        "children": <_Text
          buttonTitle={true}
          disabledButtonTitle={false}
        />,
        "onPress": [MockFunction],
        "style": Array [
          Object {
            "marginLeft": 13,
            "marginRight": 13,
          },
        ],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "buttonTitle": true,
          "children": undefined,
          "disabledButtonTitle": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <TouchableOpacity
          activeOpacity={0.2}
          onPress={[MockFunction]}
          style={
            Array [
              Object {
                "marginLeft": 13,
                "marginRight": 13,
              },
            ]
          }
        >
          <_Text
            buttonTitle={true}
            disabledButtonTitle={false}
          />
        </TouchableOpacity>,
        "style": Array [
          Object {
            "borderRadius": 15,
            "height": 30,
            "justifyContent": "center",
          },
          Object {
            "backgroundColor": "#1C497E",
          },
          undefined,
        ],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "activeOpacity": 0.2,
          "children": <_Text
            buttonTitle={true}
            disabledButtonTitle={false}
          />,
          "onPress": [MockFunction],
          "style": Array [
            Object {
              "marginLeft": 13,
              "marginRight": 13,
            },
          ],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "buttonTitle": true,
            "children": undefined,
            "disabledButtonTitle": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
